### ISSUE TITLE
nxagent -full

### ISSUE THREAD

**Issue by uli42** [ArcticaProject/nx-libs#191]
`nxagent -h` shows

`-full                  utilize full regeneration`

That switch opens the session in fullscreen for me, so the description seems a bit odd at least.

**Comment by Ionic** [comment]
Good catch, I think that should be changed to "fullscreen" then... but what exactly is the "full regeneration" and how does it toggled?!

**Comment by uli42** [comment]
`-full` comes from Xnest and is described there as

> -full
> This option tells Xnest to utilize full regeneration of real server objects and reopen a new connection to the real server each time the nested server regenerates. The sample server implementation regenerates all objects in the server when the last client of this server terminates. When this happens, Xnest by default maintains the same top-level window and the same real server connection in each new generation. If the user selects full regeneration, even the top-level window and the connection to the real server will be regenerated for each server generation.

Using this option sets `nxagentFullGeneration` and `nxagentDoFullGeneration`; both are rarely used. I doubt that we need that feature, so it probably can be dropped completely. It is missing in the manpage, BTW.

**Comment by sunweaver** [comment]
HI Uli,

On  Di 11 Jul 2017 22:02:03 CEST, Ulrich Sibiller wrote:

> `-full` comes from Xnest and is described there as
>
> ```
> -full
> This option tells Xnest to utilize full regeneration of real server  
> objects and reopen a new connection to the real server each time the  
> nested server regenerates. The sample server implementation  
> regenerates all objects in the server when the last client of this  
> server terminates. When this happens, Xnest by default maintains the  
> same top-level window and the same real server connection in each  
> new generation. If the user selects full regeneration, even the  
> top-level window and the connection to the real server will be  
> regenerated for each server generation.
> ```
>
> Using this option sets `nxagentFullGeneration` and  
> `nxagentDoFullGeneration`; both are rarely used. I doubt that we  
> need that feature, so I probably can be dropped completely.

Looks sensible. There is a lot of Xnest copy+paste code (still) in nxagent.

Will you propose a PR for this? Or shall I?

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by uli42** [comment]
Without an according commit using -full does not open a full screen for me nowadays. I am not sure what behaviour I was describing initially...

**Comment by uli42** [comment]
I will do a PR
